api/v1alpha1/broom_types.go

```diff
@@ -41,33 +41,51 @@ const (
 )
 
 type BroomAdjustment struct {
-	Type  BroomAdjustmentType `json:"type"`
-	Value string              `json:"value"`
+	Type     BroomAdjustmentType `json:"type"`
+	Value    string              `json:"value"`
+	MaxLimit string              `json:"maxLimit,omitempty"`
 }
 
-func (adj BroomAdjustment) IncreaseMemory(m *resource.Quantity) error {
+// AdjustMemory adjusts the memory based on the adjustment type, value and maxLimit.
+func (adj BroomAdjustment) AdjustMemory(m *resource.Quantity) (bool, error) {
+	before := m.DeepCopy()
 	switch adj.Type {
 	case AddAdjustment:
 		y, err := resource.ParseQuantity(adj.Value)
 		if err != nil {
-			return fmt.Errorf("unable to parse value to resource.Quantity: %w", err)
+			return false, fmt.Errorf("unable to parse value to resource.Quantity: %w", err)
 		}
 		m.Add(y)
 	case MulAdjustment:
 		y, err := strconv.Atoi(adj.Value)
 		if err != nil {
-			return fmt.Errorf("unable to parse value to int: %w", err)
+			return false, fmt.Errorf("unable to parse value to int: %w", err)
 		}
 		m.Mul(int64(y))
 	}
-	return nil
+
+	if adj.MaxLimit == "" {
+		return true, nil
+	}
+	maxLimit, err := resource.ParseQuantity(adj.MaxLimit)
+	if err != nil {
+		return false, fmt.Errorf("unable to parse maxLimit to resource.Quantity: %w", err)
+	}
+	if m.Cmp(maxLimit) == 1 { // m is greater than maxLimit
+		*m = maxLimit
+	}
+	if !m.Equal(before) {
+		return true, nil
+	}
+	return false, nil
```

Comment:
it seems to be better, i think

```suggestion
	if m.Cmp(maxLimit) == 1 { // m is greater than maxLimit
		*m = maxLimit
		return true, nil
	}
	return false ,nil
```

Reply:
I agree!

Actually, I think the meaning of the first return value (`bool`) of  this function (`AdjustMemory`) was hard to understand. Initially, I was trying to return whether the value of the memory limit "has changed from before" regardless of `Add`, `Mul` or `MaxLimit`, but there was no explanation and even I, who wrote it, was confused.

Therefore, I [changed](https://github.com/m3dev/broom/pull/8/commits/fa375aba345c06e0dae25323befcbcd0e6d2b577) to return  only `error`, and the function caller determines whether the value has been changed or not. Now it looks so simple.

Thank you!!